cpp/src/traversal/bfs_kernels.cuh

```diff
@@ -703,16 +705,21 @@ __global__ void topdown_expand_kernel(
     shared_local_new_frontier_predecessors[TOP_DOWN_BATCH_SIZE * TOP_DOWN_EXPAND_DIMX];
   __shared__ IndexType block_n_frontier_candidates;
 
-  IndexType block_offset = (blockDim.x * blockIdx.x) * max_items_per_thread;
-  IndexType n_items_per_thread_left =
-    (totaldegree - block_offset + TOP_DOWN_EXPAND_DIMX - 1) / TOP_DOWN_EXPAND_DIMX;
+  IndexType block_offset            = (blockDim.x * blockIdx.x) * max_items_per_thread;
+  IndexType n_items_per_thread_left = 0;
+
+  if (totaldegree > block_offset) {
+    n_items_per_thread_left =
+      (totaldegree - block_offset + TOP_DOWN_EXPAND_DIMX - 1) / TOP_DOWN_EXPAND_DIMX;
+  }
 
   n_items_per_thread_left = min(max_items_per_thread, n_items_per_thread_left);
 
   for (; (n_items_per_thread_left > 0) && (block_offset < totaldegree);
 
        block_offset += MAX_ITEMS_PER_THREAD_PER_OFFSETS_LOAD * blockDim.x,
-       n_items_per_thread_left -= MAX_ITEMS_PER_THREAD_PER_OFFSETS_LOAD) {
+       n_items_per_thread_left -=
+       min(n_items_per_thread_left, (IndexType)MAX_ITEMS_PER_THREAD_PER_OFFSETS_LOAD)) {
     // In this loop, we will process batch_set_size batches
     IndexType nitems_per_thread =
       min(n_items_per_thread_left, (IndexType)MAX_ITEMS_PER_THREAD_PER_OFFSETS_LOAD);
```

Comment:
Better use std::min instead of custom min.

Reply:
Done.